fr32/src/reader.rs

```diff
@@ -19,6 +19,7 @@ const NUM_U128S_PER_BLOCK: usize = NUM_BYTES_OUT_BLOCK / size_of::<u128>();
 const MASK_SKIP_HIGH_2: u128 = 0b0011_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111_1111;
 
 /// An `io::Reader` that converts unpadded input into valid `Fr32` padded output.
+#[repr(align(16))]
```

Comment:
Blocking: I believe this only aligns the struct, so to be able to rely on this, the `in_buffer` must be the first field in the struct and we need to use `repr(C)`  to avoid any reordering of the fields. 
Easier solution is to make sth like

```rust
#[repr(align(16))]
struct AlignedBuffer([u8; NUM_BYTES_IN_BLOCK + 1 ]);
```

and use that

Reply:
Ah, nice, thanks, will look at the options there.